syntax = "proto3";

option csharp_namespace = "Protos.Account";

package account;

service AccountService {
	rpc GetUserAccounts (GetUserAccountsRequest) returns (AccountsResponse);
	rpc GetAccountById (GetAccountByIdRequest) returns (AccountResponse);
	rpc	CreateAccount (AddAccountRequest) returns (AccountResponse);
	rpc UpdateAccount (UpdateAccountRequest) returns (AccountResponse);
	rpc DeleteAccount (DeleteAccountRequest) returns (AccountResponse);
}

message GetUserAccountsRequest {
	int32 userId = 1;
}

message GetAccountByIdRequest {
	int32 id = 1;
}

message AddAccountRequest {
	int32 userId = 1;
}

message UpdateAccountRequest {
	int32 id = 1;
	int32 userId = 2;
	float balance = 3;
}

message DeleteAccountRequest {
	int32 id = 1;
}

message AccountResponse {
	optional Account account = 1;
}

message AccountsResponse {
	repeated Account accounts = 1;
}

message Account {
	int32 id = 1;
	int32 userId = 2;
	float balance = 3;
}
